### DISCUSSION TITLE
Most correct way to add return_response parameter to a service

### DISCUSSION THREAD

**Discussion by TinBane** [GrandMoff100/HomeAssistantAPI#191] [Q&A] [answered]
I need to make a call to 
`/api/services/todo/get_items?return_response`
but when I use
`Domain.get_items_method(entity_id='the_entity_id'`

I get the following error:
`homeassistant_api.errors.RequestError: {"message":"Service call requires responses but caller did not ask for responses. Add ?return_response to query parameters."}`

And I can confirm from intercepting the HTTP requests that they don't have the requtn_response parameter.

But I can't work out how to add a parameter. If I add a params parameter then the trigger_service method in the rawclient.py file puts the service data under the keyword "json". There's no option to provide any keywords that will survive to the requests level.

To achieve this result, I need to take the method and set the service_id to have the parameter on the end. This seems like a really bad idea, in case it causes more problems downstream, changing something that is named an ID. Should there be a way to do this more correctly, such as taking kwargs like "params" and putting them into the request? Perhaps just allowing parameters to be supplied with methods optionally and passing them all the way to the actual request?

**Comment by GrandMoff100** [comment]
You can set `client.global_request_kwargs["params"] = {"return_responses": ""}`. This will add `?return_responses=` (equivalent to `?return_responses`) to *every* request the client makes. However, `global_request_kwargs` probably isn't documented :grimacing: 

Although, you're right that's not a good solution. I'll open an issue about it and we can continue the conversation on how to address it there.

**Comment by GrandMoff100** [comment, marked as answer]
Historically the REST API would only return the json data for states that changed while a service was triggered. But as of PR https://github.com/home-assistant/core/pull/115046 services can return data *instead* of those state changes

Upon further consideration I think in addition to being able to pass request-level parameters to service calls, it might also be worthwhile to have `return_responses` be its own built-in parameter so we can adjust the parsing logic of the service call responses.

**Reply by TinBane** [reply]
I think that would be pretty useful, certainly for my use case (trying to get items from a shopping list).

**Reply by GrandMoff100** [reply]
See #192
